public/docs/ts/latest/guide/component-communication.jade

```diff
@@ -0,0 +1,707 @@
+  
+figure.image-display
+  img(src="/resources/images/devguide/component-communication/child-to-parent.png" alt="Output event binding")
+  
++makeExample('component-communication/ts/fragments/hero-job-board-with-parentref.ts', null, 'take-job-parentref/hero-job-board.ts (extract)', {pnk: /(\[parent\]|jobBoard\(\))/g})
+
+:marked
+  `HeroPanel` now has a reference to its parent, and it can easily call the `heroTakesJob()` method:
+  
++makeExample('component-communication/ts/fragments/hero-panel-with-parentref.ts', null, 'take-job-parentref/hero-panel.ts (extract)', {blk: /(HeroJobBoard)/g, pnk: /(parent|takeJob\(\)|heroTakesJob)/g})
+
+:marked
+  Now, we can invite heroes, announce job requests, and our heroes can opt to undertake the job. The `HeroJobBoard` component lists responding
+  heroes in the UI:
+  
+figure.image-display
+  img(src="/resources/images/devguide/component-communication/take-job-ui.png" alt="Hero undertakes job")
+  
+:marked
+  ### Using `@Output`
+  
+.alert.is-critical
+  :marked
+    **`@Output()` always must be used with parentheses.** If we forget about it, our application will fail inscrutably.
+  
+:marked
+  Instead of decorating properties with the `@Output()` annotation, `@Component()` can be used to define output properties.
+  We could have declared that `onJobTaken` is an output property by setting up the `outputs` annotation property of `@Component`:
+  
++makeExample('component-communication/ts/fragments/alternative-output.ts', null, null, {pnk: /(outputs)/g})
+
+:marked
+  ### Event arguments
+  
+  A child component can easily identify its parent during parent-to-child communication, as it has exactly one parent. Thus when a parent 
+  sets an input property, the child exactly knows the source. However, when a parent component attaches an event handler method to an
+  output event of its child components, it needs to take into account that messages may be received from multiple children. It does not means
+  that the parent needs to know which of its children sends the message: nonetheless, the parent needs to know how to process the message, and
+  it may depend on its source, the child that sends it.
+  
+  When we pass arguments to events, these arguments should contain all information the parent needs to know to process the child components'
+  messages properly. We can use the non-generic `EventEmitter` type that allows using any type of event arguments. However, we'd better use
+  events with well-defined arguments that can be represented with strong TypeScript types.
+  
+.alert.is-helpful
+  :marked
+    Angular encourages us to use strong event argument types whenever it's possible, because it allows creating more maintainable and
+    less error-prone code.
+  
+:marked
+.alert.is-critical
+  :marked
+    Nothing prevents us to create event argument types that pass back a reference to the child component instance to its parent so that the 
```

Comment:
Why are we talking about this? It's not something we're showing.

To be honest, it isn't nearly as bad as passing the parent to the child.

Note also that we are passing the entire child to the parent when we use the `contentChildren` approach. If it's "bad practice" here, it must be bad practice there, right?

I think I'd just ditch this argument.